app/controllers/application_controller.rb

```diff
@@ -30,6 +30,20 @@ def upgrade_warning
     basecamp_auth_check
   end
 
+  def filtered_agent_return_link(options = {})
+    case ret = params[:return].presence || options[:return]
+      when "show"
```

Comment:
@cantino I'm not familiar with this indent style in ruby.  Is it your preference, or just that you wrote it without auto-indent?